src/main/php/com/neo4j/Serialization.class.php

```diff
@@ -0,0 +1,204 @@
+<?php namespace com\neo4j;
+
+use lang\IllegalStateException;
+use io\ByteOrder;
+
+/**
+ * Bolt protocol message serialization
+ *
+ * @see   https://boltprotocol.org/v1/#serialization
+ */
+class Serialization {
+  private static $reverse;
+
+  static function __static() {
+    self::$reverse= ByteOrder::LITTLE_ENDIAN === ByteOrder::nativeOrder();
+  }
+
+  /** Creates a marker */
+  private function marker($base, $top, $length) {
+    if ($length < 16) {
+      return pack('C', $base + $length);
+    } else if ($length < 256) {
+      return pack('CC', $top, $length);
+    } else if ($length < 65536) {
+      return pack('Cn', $top + 1, $length);
+    } else {
+      return pack('CN', $top + 2, $length);
+    }
+  }
+
+  /**
+   * Serialize a value
+   * 
+   * @param  var $value
+   * @return string
+   */
+  public function serialize($value) {
+    if (null === $value) {
+      return "\xc0";
+    } else if (true === $value) {
+      return "\xc3";
+    } else if (false === $value) {
+      return "\xc2";
+    } else if (is_int($value)) {
+      if ($value > 2147483647 || $value < -2147483648) {
+        return "\xcb".pack('NN', ($value & 0xffffffff00000000) >> 32, $value & 0x00000000ffffffff);
```

Comment:
This could be done with `pack('q', $value)` and then adjusting endianness if necessary as seen below with `pack('s', ...)`